docs/db/graphql.md

```diff
@@ -0,0 +1,217 @@
+
+```graphql
+query getGroupInfo($object: String!) {
+  group(where: {object: {name: {_eq: $object}}}) {
+    captainLogin
+    members {
+      userLogin
+    }
+  }
+}
+```
+
+Query variable:
+
+```graphql
+{"object": "ascii-art"}
+```
+
+---
+
+- The following query returns all events in a given campus, the output for this query will be the:
+  - object name
+  - path of the event
+  - parent event (this will be the parent event of the given child event)
+
+```graphql
+query eventsByCampus($campus: String!) {
+  event(where: {campus: {_eq: $campus}}) {
+    object {
+      name
+    }
+    path
+    parent {
+      path
+      object {
+        name
+      }
+    }
+  }
+}
+```
+
+Query variable:
+
+```graphql
+{ "campus": "madere" }
+```
+
+---
+
+- The following query returns information of users that are associated to an event. For the query to work it should be given two arguments : `campus` and the `object`. The output will be the user name/login, audit ratio and the event info.
+
+```graphql
+query usersEvent($campus: String!, $object: String!) {
+  event_user(where: {event: {object: {name: {_eq: $object}}, _and: {campus: {_eq: $campus}}}}) {
+    userAuditRatio
+    userLogin
+    event {
+      path
+      object {
+        name
+      }
+    }
+  }
+}
+```
+
+Query variable:
+
+```graphql
+{"campus": "pyc", "object": "Final Exam"}
+```
+
+- If we wanted to filter the users that were registered to a type of object in the event, we would just need to filter the object by `type` instead of `name`. Should look something like this:
+
+```graphql
+query usersEvent($campus: String!, $objectType: String!) {
+  event_user(where: {event: {object: {type: {_eq: $objectType}}, _and: {campus: {_eq: $campus}}}}) {
+    userAuditRatio
+    userLogin
+    event {
+      path
+      object {
+        name
+      }
+    }
+  }
+}
+```
+
+Query variable:
+
+```graphql
+{"campus": "pyc", "objectType": "raid"}
+```
+
+---
+
+- The following query returns the current `XP` of a given user, the output should be the `amount` that this user should have
+
+```graphql
+query xp($name: String!) {
+  xp(where: {user: {login: {_eq: $name}}}) {
+    user {
+      login
+    }
+    amount
+  }
+}
+```
+
+Query variable:
+
+```graphql
+{"name": "Joao"}
+```
+
+- If we wanted to filter all `XP` gain from an user from a given object, it would look something like this:
+
+```graphql
+query eventXp($userName: String!, $objectName: String!) {
+  xp_by_path(where: {user: {login: {_eq: $userName}}, _and: {object: {name: {_eq: $objectName}}}}) {
+    path
+    user {
+      login
+    }
+    event {
+      path
+    }
+    object {
+      name
+    }
+    amount
+  }
+}
+```
+
+Query variable:
+
+```graphql
+{"userName": "Joao", "objectName": "ascii-art"}
+```
+
+---
+
+- The following query gives information relevant to the onboarding games.
+
+```graphql
+query getGameInfo($name: String!) {
+  toad_result_view(where: {user: {login: {_eq: $name}}}) {
+    games
```

Comment:
this view is used for the selection dashboard. So you can precise it's the current users waiting for selection, and add `attempts`, `allowedAttempts`,  `score`, `user { login }` and the `path` or the `campus` as it could mix the different sessions from different campuses